lib/engine/game/g_18_ny/game.rb

```diff
@@ -659,7 +667,25 @@ def home_token_locations(corporation)
           @cities.reject { |c| c.available_slots.zero? }.map { |c| c.tile.hex }
         end
 
-        def tile_lay(_hex, old_tile, _new_tile)
+        def tile_lay(_hex, old_tile, new_tile)
+          if old_tile.label
+            new_color = new_tile.color
+            # add label to new tile, if this is a plain lay on a label.
+            if PLAIN_SYMBOL_HEXES.include?(new_color) &&
+               PLAIN_SYMBOL_HEXES[new_color].include?(new_tile.hex.id) &&
+               !new_tile.label
+
+              new_tile.label = old_tile.label.to_s
+            end
+
+            # remove the label when we remove a temporaily labelled tile.
+            if PLAIN_SYMBOL_HEXES.include?(old_tile.color) &&
+               PLAIN_SYMBOL_HEXES[old_tile.color].include?(new_tile.hex.id)
+
+              old_tile.label = nil
+            end
+          end
```

Comment:
If you are trying to write something generic for all titles, 674 - 687 make sense. In 18NY (and probably most games), there aren't temporarily labled tiles. Something similar to the example below would work fine since the validation was already done elsewhere (upgrades_to_correct_label?).

`new_tile.label = old_tile.label.to_s if old_tile.label && !new_tile.label`

Reply:
Yes, those interlocks work, I was just being super conservative because of how pernicious extra labelled tiles are. Will upload the simplified version you suggested. I may look at a generic thing at some point.

Reply:
Done, thanks.

Reply:
My original comment was to remove 674 - 687 and replace with the one line. Taking a look at the code, it looks like much of that is still there. Is it needed and I'm just missing something?

Reply:
I've removed the interlocks for adding a label, but you need to know which labels need to be removed from plain tiles on tile upgrade, which is why the second  block of logic is still there.

Reply:
Ahh, yes. That's what I was missing.

If you need to keep `if old_tile.label`, then you don't need to check that condition again on the new line. Otherwise, looks good.

Reply:
Done, thanks.